docs/source/edge_detection.rst

```diff
@@ -0,0 +1,245 @@
+            :align: center
+            :width: 300
+     - .. figure:: ./images/edge_detection/example_ui_default.png
+            :align: center
+            :width: 300
+
+
+Running the application with default parameters results in the following output.
+
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - Canny edges
+     - Hough Lines
+
+   * - .. figure:: ./images/edge_detection/example_canny_default.png
+            :align: center
+            :width: 300
+     - .. figure:: ./images/edge_detection/example_result_default.png
+            :align: center
+            :width: 300
+
+Sigma
+^^^^^
+
+The following figure shows the effect of increasing the ``sigma`` parameter in the Canny edge detection algorithm. For ``sigma=3`` we obtain:
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - Canny edges
+     - Hough Lines
+
+   * - .. figure:: ./images/edge_detection/example_canny_sigma3.png
+            :align: center
+            :width: 300
+     - .. figure:: ./images/edge_detection/example_result_sigma3.png
+            :align: center
+            :width: 300
+
+Note that fewer small edges were recovered by the Canny algorithm, resulting in fewer lines in the final output.
+
+Line Length
+^^^^^^^^^^^
+
+Next, we explore the effect of increasing the ``line length`` parameter in the Hough Line Transform. For ``line length=24`` we obtain:
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - With 12 pixels line length
+     - With 24 pixels line length
+
+   * - .. figure:: ./images/edge_detection/example_result_default.png
+            :align: center
+            :width: 300
+     - .. figure:: ./images/edge_detection/example_result_line24.png
+            :align: center
+            :width: 300
+
+Note that longer lines are extracted when increasing the ``line length`` parameter.
+
+Line Gap
+^^^^^^^^
+
+The following figure shows the effect of increasing the ``line gap`` parameter in the Hough Line Transform. For ``line gap=10`` we obtain:
+
+.. list-table::
+   :widths: 25 25
+   :header-rows: 1
+
+   * - With 1 pixels line gap
+     - With 4 pixels line gap
+
+   * - .. figure:: ./images/edge_detection/example_result_default.png
+            :align: center
+            :width: 300
+     - .. figure:: ./images/edge_detection/example_result_gap4.png
+            :align: center
+            :width: 300
+
+Note that increasing the ``line gap`` parameter results in increasing the number of detected lines.
+
+Threshold
+^^^^^^^^^
+
+To demonstrate the effect of the ``threshold`` parameter, we will change the detection parameters to increase the number small segments.
+The table below presents the parameters and comparative results for ``threshold=1`` and ``threshold=8``.
+
+.. list-table::
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * - Detection parameters
+     - With ``threshold=1``
+     - With ``threshold=8``
+
+   * - .. figure:: ./images/edge_detection/edge_detection_threshold.png
```

Comment:
Maybe put the parameters image above to give the side-by-side more space (hard to see).

Reply:
Done